restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1403,9 +1404,45 @@ public void onReceive(final Object message) throws Exception {
             sender.tell(switchProxy(), self);
         } else if (GetProxies.class.equals(klass)) {
             sender.tell(getProxies(message), self);
+        } else if (Hangup.class.equals(klass)) {
+            onHangup((Hangup) message, self, sender);
         }
     }
 
+    private void onHangup(Hangup message, ActorRef self, ActorRef sender) {
+        if(message.getCallDetailRecord() == null){
+            if (logger.isDebugEnabled()) {
+                logger.debug("cdr for hangup is null");
+            }
+        } else {
+            // check if it is a local or remote call.
+            if(message.getCallDetailRecord().getInstanceId().equals(RestcommConfiguration.getInstance().getMain().getInstanceId())){
+                ActorRef callTobeHangup = lookup(new GetCall(message.getCallDetailRecord().getCallPath()));
+                if (callTobeHangup != null){
+                    callTobeHangup.tell(message, self);
```

Comment:
@maria-farooq at the Conference.onStopConference() method, the Conference actor will ask all local call actors to leave.
The call actor at this point will send BYE to the sip client. 
So there is no need to send Hangup to the local call actors.

**Important!** I think there is a bug in the call actor where after Leave message is received, the call actor will never get destroyed.
As far as I can see, when the call actor receives `Leave`message:
1. Will send bye
2. Will tell MmsCallController to leave (destroy media resources)
Possible BUG -> 3. The Call.onMediaServerControllerStateChanged doesn't take care of the MmsCallController response while state is `Leaving`
We need to investigate this one

Reply:
@gvagenas do you think this is part of the scop of this PR? this bug is there regardless if we merge this patch or not no? can we handle that seperatly

Reply:
@maria-farooq `do you think this is part of the scop of this PR?` the bug is not in the scope of this PR but I believe CallManager shouldn't try to Hangup local calls.